csrc/host_ir/executor.cpp

```diff
@@ -88,6 +89,24 @@ bool HostIrExecutor::isCompiled() const {
   return (bool)host_ir_container_;
 }
 
+namespace {
+// Validates the sizes and strides of the input and output tensors
+// against the tensorviews
+void validateTensors(
+    const std::vector<at::Tensor>& tensors,
+    const std::vector<TensorView*>& tvs,
+    const ExpressionEvaluator& expr_eval) {
+  NVF_ERROR(tensors.size() == tvs.size());
+  for (const auto i : c10::irange(tensors.size())) {
+    const auto& tensor = tensors.at(i);
+    const auto& tv = tvs.at(i);
```

Comment:
FWIW, you can `zip` instead: https://github.com/NVIDIA/Fuser/blob/53f0aceed858ed8d5842de1136303c139a94326c/csrc/transform_replay.cpp#L264. 

Welcome to C++20!